From 9bdafc092f16122b9f70c402d11b70f9146654ac Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Mon, 8 Jul 2024 11:30:25 +0200
Subject: [PATCH] 0056-strcasecmp.patch

Patch-name: 0056-strcasecmp.patch
Patch-id: 56
Patch-status: |
    # Originally from https://github.com/openssl/openssl/pull/18103
    # As we rebased to 3.0.7 and used the version of the function
    # not matching the upstream one, we have to use aliasing.
    # When we eliminate this patch, the `-Wl,--allow-multiple-definition`
    # should also be removed
From-dist-git-commit: e67e9d9c40cd2cb9547e539c658e2b63f2736762
---
 crypto/o_str.c                         | 14 ++++++++++++--
 test/recipes/01-test_symbol_presence.t |  1 +
 2 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/crypto/o_str.c b/crypto/o_str.c
index dfac215..5b1d40e 100644
--- a/crypto/o_str.c
+++ b/crypto/o_str.c
@@ -338,7 +338,12 @@ int openssl_strerror_r(int errnum, char *buf, size_t buflen)
 #endif
 }
 
-int OPENSSL_strcasecmp(const char *s1, const char *s2)
+int
+#if !defined(FIPS_MODULE) && !defined(OPENSSL_SYS_UEFI)
+__attribute__ ((symver ("OPENSSL_strcasecmp@@OPENSSL_3.0.3"),
+                    symver ("OPENSSL_strcasecmp@OPENSSL_3.0.1")))
+#endif
+OPENSSL_strcasecmp(const char *s1, const char *s2)
 {
     int t;
 
@@ -348,7 +353,12 @@ int OPENSSL_strcasecmp(const char *s1, const char *s2)
     return t;
 }
 
-int OPENSSL_strncasecmp(const char *s1, const char *s2, size_t n)
+int
+#if !defined(FIPS_MODULE) && !defined(OPENSSL_SYS_UEFI)
+__attribute__ ((symver ("OPENSSL_strncasecmp@@OPENSSL_3.0.3"),
+                    symver ("OPENSSL_strncasecmp@OPENSSL_3.0.1")))
+#endif
+OPENSSL_strncasecmp(const char *s1, const char *s2, size_t n)
 {
     int t;
     size_t i;
diff --git a/test/recipes/01-test_symbol_presence.t b/test/recipes/01-test_symbol_presence.t
index 222b188..84c76d2 100644
--- a/test/recipes/01-test_symbol_presence.t
+++ b/test/recipes/01-test_symbol_presence.t
@@ -131,6 +131,7 @@ foreach (sort keys %stlibname) {
                   s| .*||;
                   # Drop OpenSSL dynamic version information if there is any
                   s|\@\@.+$||;
+                  s|\@.+$||;
                   # Return the result
                   $_
               }
-- 
2.45.4

